include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1019,6 +1033,33 @@ struct bt_bap_unicast_server_cb {
 	int (*release)(struct bt_bap_stream *stream, struct bt_bap_ascs_rsp *rsp);
 };
 
+/**
+ * @brief Register the Unicast Server.
+ *
+ * Register the Unicast Server.
+ * This will register ASCS in the GATT database.
+ *
+ * @param param  Registration parameters for ascs.
+ *
+ * @return 0 in case of success.
```

Comment:
What about in case of failure?

Reply:
I was thinking that a non-zero value is implied. The reason that I haven't specified them any further is that all errors but one will be forwarded from ascs.c, and are not defined by this function.

Reply:
Could be as easy as `0 in case of success, negative error code otherwise` just to indicate that it may return a non-0 value

Reply:
Ideally we describe all return values with `@retval` but that's not a requirement in our API guidelines so I can't enforce that ;)

Reply:
Ideally yes, I would agree. I dont like forwarding all possible errors from lower layers though, as its not part of this module per se.

But, you are right, it should definitely at the very least be '0 in case of success, negative error code otherwise'

Reply:
> Ideally yes, I would agree. I dont like forwarding all possible errors from lower layers though, as its not part of this module per se.

Indeed. What I usually do is to forward known/expected errors directly (with documentation of course), and any unknown values are changed to `ENOEXEC`. 

See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/audio/gmap.h#L193-L207

We don't have a good guideline on this in Zephyr (or Bluetooth), and we should consider starting that discussion. Just forwarding error codes through layers isn't useful in many cases, and makes it near impossible to document all return values.